### PR TITLE
Add cli-tools module

### PR DESCRIPTION
**Pull request by ppodhajski**

### CODE REVIEW

File: deploy/configs/externals/modules.yaml [hunk 1]
```diff
@@ -29,6 +29,7 @@ modules:
     whitelist:
       - arm-forge
       - blender
+      - byobu
```
Comment (pramodk) [inline deploy/configs/externals/modules.yaml]: Is this used by `end users`?
Just a note : as a developer, one can now do `spack load byobu` to load the package. It's not necessary to have module.

Comment (ppodhajski) [reply deploy/configs/externals/modules.yaml]: yes, requested by several viz team members (not spack developers)

Comment (matz-e) [reply deploy/configs/externals/modules.yaml]: I don't think that this is a good idea. These are for the end users enjoyment, not an absolute necessity. As such, I'd normally say install the binary build like for other software (neovim, starship, ytop…) into `${HOME}/.local`.

I would maybe agree to this if there is a module akin to `python-dev` that spans several of these utilities.

**Review by matz-e — CHANGES_REQUESTED** [review]

File: deploy/configs/externals/modules.yaml (hunk 1 above)
Comment (ppodhajski) [reply deploy/configs/externals/modules.yaml]: I'd not say this is merely for end-users enjoyment when everybody works on VPN and experience some connections issues.

Let's discuss it tomorrow during Deployment sync meeting, but I don't really see it feasible for non-technical people to install it themselves. Myself I have several packages/modules in my ${HOME} but the effort to train non-technical/sporadic bb5 users (on building with spack) doesn't seem to be worth it.

**Comment by ppodhajski** [comment]
following yesterday's discussion I suggest we create an umbrella module to which I'd add byobu and tmux.

Does name **cli-tools** sound good?

**Comment by matz-e** [comment]
> following yesterday's discussion I suggest we create an umbrella module to which I'd add byobu and tmux.
> 
> Does name **cli-tools** sound good?

yes

File: deploy/environments/libraries.yaml [hunk 2]
```diff
@@ -27,6 +27,7 @@ spack:
     - boost+python@1.70.0
     - boost+python@1.73.0
     - catch2
+    - cli-tools
```
Comment (matz-e) [inline deploy/environments/libraries.yaml]: If you add it to `external` modules, this is the wrong place :)

Should also really be in `externals`, shouldn't it?

**Review by matz-e — COMMENTED** [review]
Thanks! I'm nitpicking here…

File: deploy/environments/libraries.yaml (hunk 2 above)
Comment (ppodhajski) [reply deploy/environments/libraries.yaml]: indeed, sorry for overlooking it

**Review by matz-e — APPROVED** [review]
Thanks, a good start!

**Pull request merged**